src/relentless/data.py

```diff
@@ -198,18 +198,27 @@ def clear_contents(self):
             elif entry.is_dir():
                 shutil.rmtree(entry.path)
 
-    def move_contents(self, dest):
+    def move_contents(self, dest, overwrite=True):
```

Comment:
I thought we were initially planning to leave this function alone? One issue with the current `overwrite` implementation is that if the destination has some extra content (that would not be overwritten), it gets wiped out.

I propose to remove the `overwrite` option and make this function have some nicer behavior that is more typical of Unix commands, since we are looking at it now:
```
for entry in os.scandir(self.path):
    dest_entry = pathlib.Path(os.path.join(dest.path, entry.name))
    if dest_entry.exists():
        if dest_entry.is_dir() and entry.is_dir():
            shutil.copytree(entry.path, dest_entry.path, dirs_exist_ok=True)
            shutil.rmtree(entry.path)
        elif dest_entry.is_file() and entry.is_file():
            shutil.move(entry.path, dest_entry.path)
        else:
           raise OSError('Destination {} exists and does not match type of source {}'.format(dest_entry.path, entry.name))
    else:        
        shutil.move(entry.path, dest_entry.path)
```